functions/generateFilterForm.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/** Turns the types array into a pair of dropdowns for filtering
+ *
+ * @param array $types
+ * @return string
+ */
+function generateFilterForm(array $types): string{
+    $toBeDisplayed = '<form method="get"><select name="type1"><option select="selected" value="0">Type</option>';
+
+    foreach($types as $type){
+        $toBeDisplayed .= "<option value='{$type['id']}'>{$type['name']}</option>";
+    }
+    $toBeDisplayed .= "</select><select name='type2'><option select='selected' value='0'>Type</option>";
+
+    foreach($types as $type){
+        $toBeDisplayed .= "<option value='{$type['id']}'>{$type['name']}</option>";
+    }
+    $toBeDisplayed .= "</select><input type='submit' value='Filter'/></form>";
```

Comment:
Would be really cool if you pre-selected the option for any selected filters. 

For example, if I set type1 to be normal and then hit filter, the dropdown would have normal pre-selected when the page loads.